api/src/client/ApiFormationClient.ts

```diff
@@ -161,534 +156,83 @@ export const ApiFormationClient = (config: ApiConfig, logger: LogWriterType): Fo
       });
   };
 
-  const makePostBody = (
-    userData: UserData,
-    returnUrl: string,
-    config: ApiConfig,
-    foreignGoodStandingFile: InputFile | undefined
-  ): ApiSubmission => {
-    const currentBusiness = getCurrentBusiness(userData);
-    const formationFormData = currentBusiness.formationData.formationFormData;
-
-    const isManual = formationFormData.agentNumberOrManual === "MANUAL_ENTRY";
-
-    const isForeign = currentBusiness.profileData.businessPersona === "FOREIGN";
-    const isVeteranNonprofit = formationFormData.isVeteranNonprofit;
-
-    const toFormationLegalStructure = (): FormationLegalType | "veteran-nonprofit" => {
-      if (isForeign) {
-        return `foreign-${currentBusiness.profileData.legalStructureId}` as FormationLegalType;
-      } else if (isVeteranNonprofit) {
-        return "veteran-nonprofit";
-      }
-      return currentBusiness.profileData.legalStructureId as FormationLegalType;
-    };
-
-    const naicsCode =
-      currentBusiness.profileData.naicsCode.length === 6 ? currentBusiness.profileData.naicsCode : "";
-
-    const businessType = BusinessTypeMap[toFormationLegalStructure()];
+  const health: HealthCheckMethod = () => {
+    const formationFormData = generateFormationFormData(
+      {
+        agentNumberOrManual: "MANUAL_ENTRY",
+        foreignDateOfFormation: "2022/10/20",
+        foreignStateOfFormation: "Massachusetts",
+        additionalProvisions: ["provision1", "provision2"],
+        members: undefined,
+        signers: [
+          generateFormationSigner({
+            name: "test user",
+            signature: true,
+            title: "Authorized Representative",
+          }),
+        ],
+      },
+      { legalStructureId: "foreign-limited-liability-company" }
+    );
 
-    const isCorp = businessType.shortDescription === "DP";
-    const isForeignCorp = businessType.shortDescription === "FR";
-    const isDomesticNonProfit = businessType.shortDescription === ("NP" || "NV");
+    const userData = generateFormationUserData(
+      { legalStructureId: "limited-liability-company", businessPersona: "FOREIGN" },
+      {},
 
-    const isInFormAndTerms = (
-      input: "IN_BYLAWS" | "IN_FORM" | undefined,
-      terms: string | undefined
-    ): string | undefined => {
-      return input === "IN_FORM" ? terms : undefined;
-    };
+      formationFormData
+    );
 
-    // eslint-disable-next-line unicorn/consistent-function-scoping
-    const isInFormOrBylaw = (input: "IN_BYLAWS" | "IN_FORM" | undefined): "Herein" | "Bylaw" => {
-      return input === "IN_FORM" ? "Herein" : "Bylaw";
-    };
+    const returnUrl: string = "returnUrl";
+    const foreignGoodStandingFile: InputFile | undefined = undefined;
```

Comment:
[sand] If you're just setting this to undefined there's no reason to declare it. You can just leave the positional arg out of the call to `makePostBody`.

Reply:
changed